overlord/configstate/configcore/certs.go

```diff
@@ -32,6 +33,221 @@ import (
 	"github.com/snapcore/snapd/overlord/configstate/config"
 )
 
+type certificate struct {
+	Name     string
+	Path     string
+	RealPath string
+}
+
+// This function validates the name of the certificate. We only allow certificates
+// with the correct suffix ".crt" except for the ca-certificates.crt,
+// or if the name is in the blockedCerts lists.
+func isBlocked(cert certificate, blockedCerts []string) bool {
+	// Special case for ca-certificates.crt
+	if cert.Name == "ca-certificates.crt" {
+		return true
+	}
+
+	// Check that the real underlying filepath to the certificate ends with .crt
+	if !strings.HasSuffix(cert.RealPath, ".crt") {
+		return true
+	}
+
+	// See if the name appears in the blockedCerts list
+	for _, blockedCert := range blockedCerts {
+		if cert.Name == blockedCert {
+			return true
+		}
+	}
+	return false
+}
+
+// TODO Should we support recursive directories inside the certs dir?
+func getCertObjects(basePath string) ([]certificate, error) {
+	certFiles, err := ioutil.ReadDir(basePath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read base certs: %v", err)
+	}
+
+	certsObjects := []certificate{}
+	for _, cert := range certFiles {
+		if cert.IsDir() {
+			continue
+		}
+
+		// So what are we trying to achieve here? When provided with certificate directories
+		// they may or may not be symbolic links to the actual certificate file. To handle this
+		// correctly we evaluate the symbolic links and get the real path of the certificate, because
+		// we are not going to 'copy' or link to the symbolic link, instead we will recreate the symbolic
+		// link to the real path of the certificate.
+		certRealPath := filepath.Join(basePath, cert.Name())
+		if cert.Mode()&os.ModeSymlink != 0 {
+			resolvedPath, err := filepath.EvalSymlinks(certRealPath)
+			if err == nil {
+				certRealPath = resolvedPath
+			}
+		}
+
+		// If the file is not a symbolic link then Path and RealPath will be identical.
+		certObject := certificate{
+			Name:     cert.Name(),
+			Path:     filepath.Join(basePath, cert.Name()),
+			RealPath: certRealPath,
+		}
+		certsObjects = append(certsObjects, certObject)
+	}
+	return certsObjects, nil
+}
+
+// Filters out the certificates that are not allowed to be merged.
+func filterCerts(certs []certificate, blockedCerts []string) ([]certificate, error) {
+	filteredCerts := []certificate{}
```

Comment:
nitpick: I think just declaring it is more idiomatic

```suggestion
	var filteredCerts []certificate
```